tuf/api/metadata.py

```diff
@@ -675,7 +635,83 @@ def to_dict(self) -> Dict[str, Any]:
         }
 
 
-class Root(Signed):
+class _DelegatorMixin(metaclass=abc.ABCMeta):
+    """Class that implements verify_delegate() for Root and Targets"""
+
+    @abc.abstractmethod
+    def get_delegated_role(self, delegated_role: str) -> Role:
+        """Return the role object for the given delegated role.
+
+        Raises ValueError if delegated_role is not actually delegated.
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_key(self, keyid: str) -> Key:
+        """Return the key object for the given keyid.
+
+        Raises ValueError if key is not found.
+        """
+        raise NotImplementedError
+
+    def verify_delegate(
+        self,
+        delegated_role: str,
+        delegated_metadata: Metadata,
+        signed_serializer: Optional[SignedSerializer] = None,
+    ) -> None:
+        """Verify that ``delegated_metadata`` is signed with the required
+        threshold of keys for the delegated role ``delegated_role``.
```

Comment:
Off-topic and not important, but here's a chance for leaner docs, which also makes it consistent with the first part of the sentence.
```suggestion
        threshold of keys for the ``delegated_role``.
```